From 8798d62cf3329f556a2d3b7ba857585bb032f3f8 Mon Sep 17 00:00:00 2001
From: "junwei.ma" <junwei.ma@amlogic.com>
Date: Fri, 4 Aug 2023 13:47:01 +0800
Subject: [PATCH] ISP: [isp][driver] gst can't operate video50 [1/1]

PD#SWPL-134178

Problem:
gst can't operate video50

Solution:
modify driver

Verify:
ok

Signed-off-by: junwei.ma <junwei.ma@amlogic.com>
Change-Id: I4fcc4fd85721ea6ee3861f4c2b90c88f41cf5d71
---
 drivers/media/camera/ov5640.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/media/camera/ov5640.c b/drivers/media/camera/ov5640.c
index 84be52ba..bf027945 100644
--- a/drivers/media/camera/ov5640.c
+++ b/drivers/media/camera/ov5640.c
@@ -2598,7 +2598,6 @@ static int vidioc_try_fmt_vid_cap(struct file *file, void *priv,
 		field = V4L2_FIELD_INTERLACED;
 	} else if (field != V4L2_FIELD_INTERLACED) {
 		dprintk(dev, 1, "Field type invalid.\n");
-		return -EINVAL;
 	}
 
 	maxw  = norm_maxw();
@@ -2890,9 +2889,14 @@ static int vidioc_s_std(struct file *file, void *priv, v4l2_std_id i)
 static int vidioc_enum_input(struct file *file, void *priv,
 		struct v4l2_input *inp)
 {
+	if (inp->index > 1)
+		return -EINVAL;
 	inp->type = V4L2_INPUT_TYPE_CAMERA;
 	inp->std = V4L2_STD_525_60;
-	sprintf(inp->name, "Camera %u", inp->index);
+	if (inp->index)
+		snprintf(inp->name, sizeof(inp->name), "Camera %u", inp->index);
+	else
+		strlcpy(inp->name, "camera", sizeof(inp->name));
 
 	return 0;
 }
-- 
2.25.1

